libwacom/libwacom-database.c

```diff
@@ -279,8 +323,81 @@ parse_stylus_id(const char *str, WacomStylusId *id)
 	return ret;
 }
 
+static char *
+string_or_fallback (GKeyFile *keyfile, const char *group,
+		    const char *key, const char *fallback, GError *error)
+{
+	GError *local_error = NULL;
+	bool ret;
+
+	ret = g_key_file_has_key(keyfile, group, key, &local_error);
+	if (local_error) {
+		g_warning ("String fallback error: %s\n", local_error->message);
+		g_clear_error (&local_error);
+	}
+	if (ret)
+		return g_key_file_get_string(keyfile, group, key, &error);
+
+	return g_strdup(fallback);
+}
+
+static int
+int_or_fallback (GKeyFile *keyfile, const char *group,
+		 const char *key, int fallback, GError *error)
+{
+	GError *local_error = NULL;
+	bool ret;
+
+	ret = g_key_file_has_key(keyfile, group, key, &local_error);
+	if (local_error) {
+		g_warning ("int fallback error: %s\n", local_error->message);
+		g_clear_error (&local_error);
+	}
+
+	if (ret)
+		return g_key_file_get_integer(keyfile, group, key, &error);
+
+	return fallback;
+}
+
+static bool
+boolean_or_fallback (GKeyFile *keyfile, const char *group,
+		     const char *key, bool fallback, WacomStylus *stylus, GError *error)
+{
+	GError *local_error = NULL;
+	bool ret;
+
+	ret = g_key_file_has_key(keyfile, group, key, &local_error);
+	if (local_error) {
+		g_warning ("boolean fallback error: %s\n", local_error->message);
+		g_clear_error (&local_error);
+	}
+
+	if (ret)
+		return g_key_file_get_boolean(keyfile, group, key, &error);
+
+	return fallback;
+}
+
+static gchar **
+garray_to_string_list(GArray *array)
+{
+	WacomStylusId *id;
+
+	if (array == NULL)
+		return NULL;
```

Comment:
```suggestion
	if (array == NULL || array->len == 0)
		return NULL;
```
to avoid
```
not ok /load/0000:0000 - GLib-FATAL-CRITICAL: g_array_new: assertion 'elt_size > 0' failed
```